runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -988,8 +988,8 @@ J9::Power::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR::Register *cndReg = cg->allocateRegister(TR_CCR);
    generateTrg1MemInstruction(cg, TR::InstOpCode::Op_load, node, scratchReg,
                      new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetInDataBlock, TR::Compiler->om.sizeofReferenceAddress(), cg));
-   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, scratchReg, scratchReg, cndReg, -1);
-   generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, unresolvedLabel, cndReg);
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::cmpi4, node, cndReg, scratchReg, -1);
```

Comment:
Since we're reading the value of `scratchReg` as an address-sized value using `Op_load`, shouldn't this be an address-sized compare (i.e. `Op_cmpi`)?

Reply:
I think you're right @aviansie-ben. On Z it is an address-sized compare as well:
https://github.com/eclipse/openj9/blob/9ef399f4f32fdf48b8bbae6bacd1ac3949e73940/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L3748-L3757

Good catch!

Reply:
Right. I was thinking of the offset being 32 bits longs. This will be changed to the 64 bit flavor `Op_cmpi`.